rest/config.go

```diff
@@ -461,6 +504,76 @@ func (dbConfig *DbConfig) validateSgDbConfig() error {
 
 }
 
+// Checks for deprecated cache config options and if they are set it will return a warning. If the old one is set and
+// the new one is not set it will set the new to the old value. If they are both set it will still give the warning but
+// will choose the new value.
+func (dbConfig *DbConfig) deprecatedConfigCacheFallback() (warnings []string) {
+
+	warningMsgFmt := "Using deprecated config option: %q. Use %q instead."
+
+	if dbConfig.DeprecatedRevCacheSize != nil {
+		if dbConfig.CacheConfig == nil || dbConfig.CacheConfig.RevCacheConfig == nil || dbConfig.CacheConfig.RevCacheConfig.Size == nil {
+			dbConfig.CacheConfig.RevCacheConfig.Size = dbConfig.DeprecatedRevCacheSize
+		}
+		warnings = append(warnings, fmt.Sprintf(warningMsgFmt, "rev_cache_size", "cache.rev_cache.size."))
+	}
+
+	if dbConfig.CacheConfig != nil {
+
+		if dbConfig.CacheConfig.DeprecatedCachePendingSeqMaxWait != nil {
+			if dbConfig.CacheConfig.ChannelCacheConfig == nil || dbConfig.CacheConfig.ChannelCacheConfig.MaxWaitPending == nil {
+				dbConfig.CacheConfig.ChannelCacheConfig.MaxWaitPending = dbConfig.CacheConfig.DeprecatedCachePendingSeqMaxWait
+			}
+			warnings = append(warnings, fmt.Sprintf(warningMsgFmt, "max_wait_pending", "channel_cache.max_wait_pending"))
```

Comment:
Do we need to prefix the new options of these warnings with `cache.`?

Reply:
I omitted this so it was relative to where the old option was vs the new but makes more sense to have the full path.